flytekit/types/file/file.py

```diff
@@ -693,16 +730,25 @@ async def async_to_python_value(
 
         # For the remote case, return an FlyteFile object that can download
         local_path = ctx.file_access.get_random_local_path(uri)
-
-        def _downloader():
-            return ctx.file_access.get_data(uri, local_path, is_multipart=False)
-
         expected_format = FlyteFilePathTransformer.get_format(expected_python_type)
-        ff = FlyteFile.__class_getitem__(expected_format)(local_path, _downloader)
+        ff = FlyteFile.__class_getitem__(expected_format)(
+            path=local_path, downloader=lambda: self.downloader(remote_path=uri, local_path=local_path)
+        )
         ff._remote_source = uri
 
         return ff
 
+    @staticmethod
+    def downloader(remote_path: str, local_path: str) -> None:
```

Comment:
Could we use the context that we pass to async_to_python_value?
```suggestion
    def downloader(ctx: FlyteContext, remote_path: str, local_path: str) -> None:
```

Reply:
Perfect! 

We could just reuse the passed context instead of getting it again from `FlyteContextManager`.